utils/model.py

```diff
@@ -0,0 +1,70 @@
+import torch
+import torch.nn as nn
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+class BaseModel(nn.Module, ABC):
+    @abstractmethod
+    def forward(self, x):
+        pass
+
+    @abstractmethod
+    def fit(self, train_loader, val_loader, num_epochs, learning_rate, device):
+        pass
+
+    @abstractmethod
+    def predict(self, x):
+        pass
+
+
+def train_model(model, train_loader, val_loader, num_epochs, learning_rate, device, save_path):
```

Comment:
We can leave this completely to model_specific `model.py`. If one model wants, it can inherit from other models.